python/ray/train/_internal/backend_executor.py

```diff
@@ -245,32 +250,96 @@ def _share_cuda_visible_devices(self):
             - Worker2: "0,1"
 
         """
-
         node_ids_and_gpu_ids = [
-            (w.metadata.node_id, w.metadata.gpu_ids) for w in self.worker_group.workers
+            (w.metadata.node_id, w.metadata.gpu_and_accelerator_ids[ray_constants.GPU])
+            for w in self.worker_group.workers
+        ]
+        self._share_runtime_ids(
+            node_ids_and_runtime_ids=node_ids_and_gpu_ids,
+            env_var=ray_constants.CUDA_VISIBLE_DEVICES_ENV_VAR,
+        )
+
+    def _share_neuron_core_ids(self):
+        """Sets NEURON_RT_VISIBLE_CORES on all workers.
+
+        For each worker, NEURON_RT_VISIBLE_CORES will be set to the
+        NEURON_CORE IDs visible to all workers on that worker's node.
+
+        This allows the workers on the same node to communicate
+        with one another.
+
+        Example:
+
+            Setup:
+            - Node1:
+                - Worker1: {0, 1}
+                - Worker2: {2, 3}
+            - Node2:
+                - Worker3: {0, 1}
+
+            NEURON_RT_VISIBLE_CORES:
+            - Worker1: "0,1,2,3"
+            - Worker2: "0,1,2,3"
+            - Worker2: "0,1"
+        """
+        node_ids_and_neuron_core_ids = [
+            (
+                w.metadata.node_id,
+                w.metadata.gpu_and_accelerator_ids[ray_constants.NEURON_CORES],
+            )
+            for w in self.worker_group.workers
         ]
+        self._share_runtime_ids(
+            node_ids_and_runtime_ids=node_ids_and_neuron_core_ids,
+            env_var=ray_constants.NEURON_RT_VISIBLE_CORES_ENV_VAR,
+        )
 
+    def _share_runtime_ids(
+        self, node_ids_and_runtime_ids: List[Tuple[str, List[str]]], env_var: str
+    ):
```

Comment:
Can you rename `runtime_ids` to `resource_ids` to match the `get_resource_ids()` naming convention?